File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/shrines/AbundantShrineBlock.java
package generations.gg.generations.core.generationscore.common.world.level.block.shrines;

import generations.gg.generations.core.generationscore.common.util.GenerationsUtils;
import generations.gg.generations.core.generationscore.common.world.item.GenerationsItems;
import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntities;
import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntityModels;
import generations.gg.generations.core.generationscore.common.world.level.block.entities.shrines.AbundantShrineBlockEntity;
import net.minecraft.server.level.ServerPlayer;
import net.minecraft.world.level.block.state.BlockBehaviour;

import java.util.function.Consumer;

public class AbundantShrineBlock extends ShrineBlock<AbundantShrineBlockEntity> {

    private static final Consumer<ServerPlayer> playerConsumer = p -> GenerationsUtils.giveItem(p, GenerationsItems.REVEAL_GLASS.get().getDefaultInstance());

    public AbundantShrineBlock(BlockBehaviour.Properties properties) {
        super(properties, GenerationsBlockEntities.ABUNDANT_SHRINE, GenerationsBlockEntityModels.ABUNDANT_SHRINE);
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/shrines/AbundantShrineBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block.shrines

import com.mojang.serialization.MapCodec
import generations.gg.generations.core.generationscore.common.util.GenerationsUtils
import generations.gg.generations.core.generationscore.common.world.item.GenerationsItems
import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntities
import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsBlockEntityModels
import net.minecraft.server.level.ServerPlayer
import java.util.function.Consumer

class AbundantShrineBlock(properties: Properties) : ShrineBlock(
    properties,
    GenerationsBlockEntityModels.ABUNDANT_SHRINE
) {
    override val blockEntityType
        get() = GenerationsBlockEntities.ABUNDANT_SHRINE

    override fun codec(): MapCodec<AbundantShrineBlock> = CODEC

    companion object {
        private val playerConsumer = Consumer { p: ServerPlayer -> GenerationsUtils.giveItem(p, GenerationsItems.REVEAL_GLASS.value().defaultInstance) }
        val CODEC = simpleCodec(::AbundantShrineBlock)
    }
}
